### PR TITLE
feat(in-app-messaging): add React Native demo app

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add React Native In-App Messaging `Demo` and components:
- restructure example app, adding _App_, _assets_, _ui_, and _features_ directories
- add `useInAppDemo` hook ([see](https://github.com/aws-amplify/amplify-ui/blob/in-app-messaging/main/examples/next/pages/ui/components/in-app-messaging/demo/utils.ts))
- add `Checkbox`, `Label`, `Radio`, and `RadioGroup` base components (these are not production ready but are meant to be completed and moved to _@aws-amplify/ui-react-native_ once completed)
- prevent `LogBox` from printing a "require cycle" warning from `storybook` that is should be fixed in the next major version
- update `IconButton` props to include `iconStyle` prop and add corresponding unit test

image

#### Issue #, if available
NA

#### Description of how you validated changes
Manually tested
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/react-native/src/ui/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Re-export internal React Native UI primitives and types for use in example apps and storybook.
+ * This file should be removed when primitves are exposed for external usage
```
Comment (joebuono) [inline examples/react-native/src/ui/index.ts:3]: nit: "primitives"

**Review by joebuono — APPROVED** [review]
LGTM, nothing blocking. Perhaps the primary and secondary buttons can be slightly different colors?

**Comment by calebpollman** [comment]
> LGTM, nothing blocking. Perhaps the primary and secondary buttons can be slightly different colors?

The button styling is based on the message payload defaults, and this demo app does not cover custom styling, but we should definitely build a custom styled example app as well 😄

**Review by slaymance — APPROVED** [review]
LGTM!

**Pull request merged**